modPrecompiled/test/org/aion/precompiled/contracts/EDVerifyContractTest.java

```diff
@@ -0,0 +1,138 @@
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Centrys
+ */
+
+package org.aion.precompiled.contracts;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.aion.base.type.Address;
+import org.aion.base.type.IExecutionResult;
+import org.aion.crypto.ECKey;
+import org.aion.crypto.ECKeyFac;
+import org.aion.crypto.HashUtil;
+import org.aion.crypto.ISignature;
+import org.aion.mcf.vm.types.DataWord;
+import org.aion.precompiled.ContractFactory;
+import org.aion.vm.*;
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.spongycastle.util.encoders.Hex;
+
+public class EDVerifyContractTest {
+
+    private byte[] txHash = RandomUtils.nextBytes(32);
+    private Address origin = Address.wrap(RandomUtils.nextBytes(32));
+    private Address caller = origin;
+
+    private Address blockCoinbase = Address.wrap(RandomUtils.nextBytes(32));
+    private long blockNumber = 1;
+    private long blockTimestamp = System.currentTimeMillis() / 1000;
+    private long blockNrgLimit = 5000000;
+    private DataWord blockDifficulty = new DataWord(0x100000000L);
+
+    private DataWord nrgPrice;
+    private long nrgLimit;
+    private DataWord callValue;
+    private byte[] callData;
+
+    private int depth = 0;
+    private int kind = ExecutionContext.CREATE;
+    private int flags = 0;
+
+    @Before
+    public void setup() {
+        nrgPrice = DataWord.ONE;
+        nrgLimit = 20000;
+        callValue = DataWord.ZERO;
+        callData = new byte[0];
+        TransactionResult txResult = new TransactionResult();
+    }
+
+    @Test
+    public void shouldReturnSuccessTestingWith256() {
+        ECKeyFac.setType(ECKeyFac.ECKeyType.ED25519);
+        ECKey ecKey = ECKeyFac.inst().create();
+        ecKey = ecKey.fromPrivate(Hex.decode(
+            "5a90d8e67da5d1dfbf17916ae83bae04ef334f53ce8763932eba2c1116a62426fff4317ae351bda5e4fa24352904a9366d3a89e38d1ffa51498ba9acfbc65724"));
+
+        byte[] pubKey = ecKey.getPubKey();
+
+        byte[] data = "Our first test in AION1234567890".getBytes();
+
+        HashUtil.setType(HashUtil.H256Type.KECCAK_256);
+        byte[] hashedMessage = HashUtil.h256(data);
+
+        ISignature signature = ecKey.sign(hashedMessage);
+
+        byte[] input = new byte[128];
+        System.arraycopy(hashedMessage, 0, input, 0, 32);
+        System.arraycopy(signature.getSignature(), 0, input, 32, 64);
+        System.arraycopy(pubKey, 0, input, 96, 32);
+
+        ExecutionContext ctx = new ExecutionContext(txHash,
+            ContractFactory.getEdVerifyContractAddress(), origin, caller, nrgPrice,
+            nrgLimit, callValue,
+            callData, depth, kind, flags, blockCoinbase, blockNumber, blockTimestamp, blockNrgLimit,
+            blockDifficulty);
+        IPrecompiledContract contract = new ContractFactory().getPrecompiledContract(ctx, null);
+
+        assert contract != null;
```

Comment:
can we use assert() or assertThat() instead ?

Reply:
sure. it's suggested by IDE.